airflow/jobs/backfill_job.py

```diff
@@ -621,6 +621,9 @@ def to_keep(key: TaskInstanceKey) -> bool:
 
                 ti_status.to_run.update({ti.key: ti for ti in mapped_tis})
 
+                for new_ti in mapped_tis:
+                    new_ti.set_state(TaskInstanceState.SCHEDULED, session=session)
```

Comment:
It’s probably moot since BackfillJob is a mess, but why don’t we do this in `_manage_executor_state` directly when `mapped_tis` are created?

Reply:
Yeah, this is hack on top of hack. I'm going to schedule re-writting Backfill from the ground up for 2.4/2.5.

Reply:
Looking forward to it